Makefile

```diff
@@ -202,6 +202,16 @@ format:
 	@./tools/format_all.sh
 	@make --no-print-directory -C test format
 
+.PHONY: typos
+typos:
```

Comment:
I am not sure that `typos` deserves a Makefile target. The `cargo format` and `cargo clippy` do not have their own Makefile targets. And this is by design: the users of this Makefile only need to run a single command `make check` to detect all sorts of minor issues in the codebase at compile time.